integration_tests/rebalance_test.go

```diff
@@ -124,6 +125,111 @@ func (s *IntegrationTestSuite) TestRebalance() {
 			return true
 		}, 100*time.Second, 1*time.Second, "rebalance request took too long")
 
+		s.T().Logf("wait for new vote period start")
```

Comment:
Wait, does this actually work? If you are waiting until the next vote period after steward has submitted the precommits/commits they should be gone already.

Reply:
It is technically a race but it works in my experience because the next vote period is ~15-20 seconds away when the rebalance request gets sent (at least on my machine). I feel like that is enough room to call it safe.. you? Can you run it and observe the steward debug logs to make sure you see the same?

Reply:
I think this is only actually working because everything is submitted fast enough that it's in the same block, which will still be a vote period start. I would remove that delay bit, steward waits for the vote period start inside `decide_rebalance`.